src/components/cards/LearningModule.tsx

```diff
@@ -0,0 +1,80 @@
+import { Fragment, useEffect, useState } from "react";
+import ArrowButton from "@/components/ui/button/Arrow";
+import Link from "next/link";
+import { useTranslation } from "next-i18next";
+import Tag from "../ui/Tag";
+import { DurationCard } from "@/components/badges/Duration";
+import { useMultiSelector } from "@/hooks/useTypedSelector";
+import { IRootState } from "@/store";
+import { LearningModule } from "@/types/course";
+
+/**
+ * Props for the RelatedLearning component.
+ */
+
+/**
+ * Component that displays related learning material with a title, description, and "Start now" button.
+ */
+export function LearningModuleCard({ data }: { data: LearningModule }): JSX.Element {
+  const [level, setLevel] = useState("");
+  const { t } = useTranslation()
+  const { challenge, community, colors } = useMultiSelector<any, any>({
+    challenge: (state: IRootState) => state.challenges.current,
+    community: (state: IRootState) => state.communities.current,
+    colors: (state: IRootState) => state.ui.colors
+  })
+
+  useEffect(() => {
+    if (!challenge?.level) return
+    if (challenge.level === 0 || challenge.level === 1) return setLevel("course.challenge.level-0");
+    return setLevel("course.challenge.level-2");
+  }, [challenge?.level]);
+
```

Comment:
use a Memo since a state is only updated in one place and that place is a useEffect

```ts
const level = useMemo(()=> { 
     if (!challenge?.level) return "";
     if (challenge?.level === 0 || challenge?.level === 1) return "course.challenge.level-0";
     return "course.challenge.level-2"
 }, [challenge?.level]);
 ```